pie.py

```diff
@@ -230,9 +228,10 @@ def enter(cls,ctx):
     def cmd(cls,c,i=None):
         if i is None: i=len(cls.context)
         if i>0: return cls.context[i-1].cmd(c)
-        errorcode=CMD_FN(c,shell=True)
-        if errorcode!=0:
-            raise cls.CmdError(errorcode,c)
+        process = subprocess.Popen(c, shell=True, stdout=sys.stdout, stderr=sys.stderr, universal_newlines=True)

```

Comment:
Was `universal_newlines` needed @roleyfoley? It changes the stdout and stderr streams from binary to text, which can make it susceptible to encoding errors. I'm just concerned with breaking something for other people using pie.

Reply:
It was taking the tidiness to the next level. The exception message was all on a single line when I was testing it. Can remove it to maintain compatibility

Reply:
Ah, was it - sorry, I just couldn't test the impact easily. Don't worry about this PR, I'm re-writing it a bit so that it's still overrideable as the CMD_FN is used to mock and possibly in real world uses of the code too to change the behaviour.